### DISCUSSION TITLE
Fix UI for Linux users

### DISCUSSION THREAD

**Discussion by smgrey** [cabarius/ToyBox#1089] [General]
If you use Linux and you are having issues with empty UI elements, either all of them or just check boxes, then what you need to do is to install and select a proper font. I only know of 3 fonts at the moment that work: any of the Meiryo fonts and Unifont that you can install using winetricks, or Meslo NerdFont that you can download and just copy into the fonts folder of your wine prefix. After installing any of these fonts, go to the ToyBox settings and select the font there, click "Apply" and you should have a working ToyBox.

P.S. I think it would be a good idea for maintainers of the project to either add a default font with all the glyphs into the project, or move the glyphs out into a config file (perhaps localization) so that someone could change them and use the font with no glyphs.

**Comment by xADDBx** [comment]
Hmm. a few updates ago I added a fallback font which should be used if the normal one isn't available. Which game + version of ToyBox did you try?

**Reply by xADDBx** [reply]
Just looked it up; seems like I only added that change to RT and didn't back port for Wrath?

**Reply by smgrey** [reply]
I am using it for Pathfinder WOTR at the moment (version 2.2.4p.1022_release_(70529)) and when I installed the ToyBox first, I had no text anywhere in the UI elements until I manually installed the font. Then it took me some time to figure out that the glyphs are a font problem as well (I wasn't sure if it's a rendering issue or just font didn't have the glyphs in question) and I had to try many fonts until the 3 I mentioned worked. No other fonts showed checkmarks and crosses in checkboxes, plus the arrow in the dropdown boxes didn't display correctly.

**Comment by xADDBx** [comment]
Here's a version with the Glyph Stuff backported:

[0ToyBox0-Wrath.zip](https://github.com/cabarius/ToyBox/files/15050394/0ToyBox0-Wrath.zip)
To quote myself from another post:
> 
>     * After starting and opening the mod GUI a check is done whether the current system supports glyphs; if not they will be disabled after restarting the game
> 
>     * Added an Override Option to the Settings which will allow disabling said check (and after that choosing whether to disable or enable default glyphs)
> 
> 
> It works on my system but I don't have a system which doesn't support Glyphs Relevant changes are in [85b060c](https://github.com/xADDBx/ToyBox/commit/85b060ca471a4b7638cf549cbcd44ce2f7798a9f)

**Reply by smgrey** [reply]
Hmmm, this didn't work for me, is this the main version right now? I will keep it as as backup. After I installed font with all glyphs, I have no issues with the ToyBox. I am using Meiryo UI font now and I can recommend it as the default font. Meslo NerdFont is also very good and there are a lot of different versions of them.
